### PR TITLE
Override Default Tenant Limits via Env Vars

### PR DESCRIPTION
**Pull request by barrownicholas**
## What kind of change does this PR introduce?

Feature: in continuation of the conversation [here](https://github.com/supabase/realtime/pull/866#issuecomment-2176530588) with @filipecabaco , this PR allows overriding a host of the default limits (used when creating tenants for the first time) using environment variables. This will greatly simplify self-hosting initial-setup. 

## What is the current behavior?

Tenant default values are hard-coded and need to be manually changed **after** the tenant is created, making for a more difficult onboarding process for self-hosted users. 

## What is the new behavior?

Specifying certain `TENANT_...` environment variables overrides the default values whenever a tenant is created. 

## Additional context

This is just an extension of the solution used in #843 and #866

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -4,7 +4,7 @@ defmodule Realtime.MixProject do
   def project do
     [
       app: :realtime,
-      version: "2.29.10",
+      version: "2.30.0",
```
Comment (barrownicholas) [inline mix.exs:7]: I went with a `minor` bump instead of a `patch` since this was a bit more substantative, but happy to use `2.29.11` if that's better

Comment (filipecabaco) [reply mix.exs:7]: lets go with `2.29.11` for now as all of this changes are non-breaking additive elements

**Comment by barrownicholas** [comment]
@filipecabaco I bumped the version back to `2.29.11` so this should be good to go when you have a sec to look at it

**Review by filipecabaco — APPROVED** [review]

**Pull request merged**